kolibri/core/device/utils.py

```
@@ -141,3 +111,219 @@ def set_app_key_on_response(response):
+
+        if username and password and facility:
+            try:
+                FacilityUser.objects.create_superuser(
+                    username, password, facility=facility
+                )
+                logger.info(
+                    "Superuser created with username '{username}' in facility '{facility}'.".format(
+                        username=username, facility=facility
+                    )
+                )
+            except ValidationError:
+                logger.warn(
+                    "An account with username '{username}' already exists in facility '{facility}', not creating user account.".format(
+                        username=username, facility=facility
+                    )
+                )
+
+
+def provision_from_file(file_path):
+    """
+    Expects a JSON file with the following format (example values supplied):
+    {
+        "facility": "My Facility",
+        "preset": "formal",
+        "facility_settings": {
+            "learner_can_edit_username": true,
+            "learner_can_edit_name": true,
+            "learner_can_edit_password": true,
+            "learner_can_sign_up": true,
+            "learner_can_delete_account": true,
+            "learner_can_login_with_no_password": true,
+            "show_download_button_in_learn": true
+        },
+        "device_settings": {
+            "language_id": "en",
+            "landing_page": "homepage",
+            "allow_guest_access": true,
+            "allow_peer_unlisted_channel_import": true,
+            "allow_learner_unassigned_resource_access": true,
+            "name": "My Device",
+            "allow_other_browsers_to_connect": true
+        },
+        "username": "superuser",
+        "password": "password"
+    }
+    All fields are optional.
+    """
+    from kolibri.core.auth.models import Facility
+
+    if device_provisioned() or not os.path.exists(file_path):
+        return
```

for a function called `provision_from_file`, these conditions seem like they should be exceptions, not silently fail

---

Fair - it was just simpler to fail silently, but can catch the error where it's invoked instead.